e2e/testcases/reconciler_finalizer_test.go

```diff
@@ -216,14 +236,22 @@ func TestReconcilerFinalizer_MultiLevelForeground(t *testing.T) {
 	deployment1NN := types.NamespacedName{Name: "helloworld-1", Namespace: repoSyncNN.Namespace}
 	deployment2NN := types.NamespacedName{Name: "helloworld-2", Namespace: repoSyncNN.Namespace}
 	namespace1NN := types.NamespacedName{Name: repoSyncNN.Namespace}
+	safetyNamespace1NN := types.NamespacedName{Name: rootRepo.SafetyNSName}
+	safetyNamespace2NN := types.NamespacedName{Name: nsRepo.SafetyNSName}
 
 	nt.T.Cleanup(func() {
-		cleanupMultiLevel(nt,
-			rootSyncNN,
-			repoSyncNN,
-			deployment1NN,
-			deployment2NN,
-			namespace1NN)
+		cleanupSyncsAndObjects(nt,
+			[]client.Object{
+				fake.RootSyncObjectV1Beta1(rootSyncNN.Name),
+				fake.RepoSyncObjectV1Beta1(repoSyncNN.Namespace, repoSyncNN.Name),
+			},
+			[]client.Object{
+				fake.DeploymentObject(core.Name(deployment1NN.Name), core.Namespace(deployment1NN.Namespace)),
+				fake.DeploymentObject(core.Name(deployment2NN.Name), core.Namespace(deployment2NN.Namespace)),
+				fake.NamespaceObject(namespace1NN.Name),
+				fake.NamespaceObject(safetyNamespace1NN.Name),
+				fake.NamespaceObject(safetyNamespace2NN.Name),
+			})
```

Comment:
These lines are also in TestReconcilerFinalizer_MultiLevelMixed. Put them in a helper func.

Reply:
done